nodeup/pkg/model/encryption.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package model
+
+import (
+	"fmt"
+
+	"k8s.io/apiserver/pkg/server/options/encryptionconfig"
+	"k8s.io/kops/pkg/apis/kops"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/nodeup/nodetasks"
+
+	"github.com/golang/glog"
+)
+
+// EncryptionBuilder builds the encryption-at-rest config
+type EncryptionBuilder struct {
+	*NodeupModelContext
+}
+
+var _ fi.ModelBuilder = &EncryptionBuilder{}
+
+// Build is responsible for building the config file
+func (b *EncryptionBuilder) Build(c *fi.ModelBuilderContext) error {
+	glog.Info("Starting encryption builder.")
+	config, err := b.buildEncryptionConfig()
+	if err != nil {
+		return err
+	}
+	manifest, err := kops.ToRawYaml(config)
+	if err != nil {
+		return fmt.Errorf("error marshalling authentication config to yaml: %v", err)
+	}
+
+	t := &nodetasks.File{
+		Path:     *b.Cluster.Spec.KubeAPIServer.ExperimentalEncryptionProviderConfig,
+		Contents: fi.NewBytesResource(manifest),
+		Type:     nodetasks.FileType_File,
+	}
+	c.AddTask(t)
+	return nil
+}
+
+func (b *EncryptionBuilder) buildEncryptionConfig() (*encryptionconfig.EncryptionConfig, error) {
+	resources := b.Cluster.Spec.KubeEtcdEncryptionConfig.Resources
+	config := &encryptionconfig.EncryptionConfig{
```

Comment:
this could probably be a oneliner 

```go
return &encryptionconfig.EncryptionConfig{
  Kind:       "EncryptionConfig",
  APIVersion: "v1",
  Resources:  b.Cluster.Spec.KubeEtcdEncryptionConfig.Resources,
}, nil
```